ground/src/main/java/com/google/android/ground/ui/common/LocationOfInterestHelper.kt

```diff
@@ -17,46 +17,58 @@ package com.google.android.ground.ui.common
 
 import android.content.res.Resources
 import com.google.android.ground.R
-import com.google.android.ground.model.AuditInfo
-import com.google.android.ground.model.User
-import com.google.android.ground.model.geometry.LineString
-import com.google.android.ground.model.geometry.LinearRing
+import com.google.android.ground.model.geometry.Geometry
 import com.google.android.ground.model.geometry.MultiPolygon
 import com.google.android.ground.model.geometry.Point
 import com.google.android.ground.model.geometry.Polygon
 import com.google.android.ground.model.locationofinterest.LocationOfInterest
-import java.util.Optional
 import javax.inject.Inject
 
-/** Common logic for formatting attributes of [LocationOfInterest] for display to the user. */
+/** Helper class for creating user-visible text. */
 class LocationOfInterestHelper @Inject internal constructor(private val resources: Resources) {
-  fun getCreatedBy(locationOfInterest: Optional<LocationOfInterest>): String =
-    getUserName(locationOfInterest).map { resources.getString(R.string.added_by, it) }.orElse("")
 
-  // TODO(#793): Allow user-defined LOI names for other LOI types.
-  fun getLabel(locationOfInterest: Optional<LocationOfInterest>): String =
-    locationOfInterest.map(::getLabel).orElse("")
+  fun getDisplayLoiName(loi: LocationOfInterest): String {
+    val loiId = loi.customId.ifEmpty { loi.getProperty("id") }
+    val loiName = loi.getProperty("name")
 
-  fun getLabel(loi: LocationOfInterest): String {
-    // TODO(#2046): Reuse logic from card util to display LOI label
-    val caption = loi.customId?.trim { it <= ' ' } ?: ""
-    return caption.ifEmpty { getLocationOfInterestType(loi) }
+    return if (loiName.isNotEmpty() && loiId.isNotEmpty()) {
+      "$loiName ($loiId)"
+    } else if (loiName.isNotEmpty()) {
+      loiName
+    } else if (loiId.isNotEmpty()) {
+      "${loi.geometry.toType()} ($loiId)"
+    } else {
+      loi.geometry.toDefaultName()
+    }
```

Comment:
how about going with `when`, the nesting of if else will keep growing in the future.

WDYT?

Reply:
I'm not sure how to achieve that since `when` expects a statement or an expression.

Reply:
```
when {
        loiName.isNotEmpty() -> ...
        else -> ...
    }
```

This way we can use when, it's doable.

Reply:
Thanks! Done